### PR TITLE
Include WRFCHEM workflow and update the GSI section for NYS Mesonet surface measurement assimilation

### PR DESCRIPTION
**Pull request by weiwilliam**
This pull request contains the updates below for issue #9 and #10:
- The assimilation of NYS mesonet
- Remove the realtime folder under $rundir/gsi
- Update the diagnostic files from binary to NetCDF for easier access
- Move the call of create_mockdata.sh into the cycle loop
- Change the symbolic link name of observation and GSI section in create_case.sh
- Update the store_case.sh to new structure of GSI running folder
- Add the data prepare scripts

### CODE REVIEW

File: run_wrfgsi.bash [hunk 1]
```diff
@@ -37,18 +42,21 @@ elif [ $realtime -eq 0 ]; then
...
    if [ $LISTOS -eq 1 ]; then
      num_metgrid_levels=32
      gfssource="/network/rit/lab/josephlab/LIN/WORK/DATA/WRF-ICBC/GFS_180714_180817"
-     gdassource="/network/rit/lab/josephlab/LIN/WORK/DATA/GSI-OBS/201808"
-     datpath="/network/asrc/scratch/lulab/WRF-GSI-CASE/mockdata"
-     obsdir="/network/asrc/scratch/lulab/WRF-GSI-CASE/mockdata/logs"
-     sh $syspath/create_mockdata.bash $gfssource $gdassource $datpath $obsdir $sdate $syspath
+     gdassource="/network/rit/lab/josephlab/LIN/WORK/DATA/GSI-OBS/summer"
+     datpath="${runpath}/mockdata"
+     obsdir= "${runpath}/mockdata/logs"
```
Comment (dpgrogan27) [inline run_wrfgsi.bash:57]: remove space after =

**Comment by weiwilliam** [comment]
The WRF-Chem upgrade from #3 has been included in the PR as well.

**Comment by Hui4ying3LUO2** [comment]
Correct me if I'm wrong: the master branch is now updated by the swei_gsi-nysm we updated last week (after deleting the extra space after =) and runs DA of both gdas and surface mesonet. Could you please add options to choose data for DA, mostly for the sake of retro studies? Thanks!

**Comment by weiwilliam** [comment]
@Hui4ying3LUO2 Yup, the master branch has been updated, but the space is still there.
Normally, we need to modify the problems during the test and update this branch before merging.
@dpgrogan27 also encountered some issues, we can fix them and include some features you mentioned for v1.1.

**Pull request merged**